engine/src/main/java/org/hibernate/search/engine/common/impl/SearchIntegrationBuilderImpl.java

```diff
@@ -194,11 +199,14 @@ public SearchIntegrationPartialBuildState prepareBuild() {
 
 			threadProviderHolder = THREAD_PROVIDER.getAndTransform( propertySource, beanResolver::resolve );
 			ThreadPoolProviderImpl threadPoolProvider = new ThreadPoolProviderImpl( threadProviderHolder );
+			engineThreads = new EngineThreads( threadPoolProvider );
+			timingSource = new DefaultTimingSource( engineThreads );
```

Comment:
There's a `SuppressingCloser` below; you should probably close these with that closer?

Reply:
Yes the idea was to have the variables declared in the outer block for that... but I forget to add the code lines to close them on error on startup...